backend/src/controllers/__tests__/player.controller.test.ts

```diff
@@ -0,0 +1,204 @@
+import * as PlayerService from "../../services/player.service";
+import { createPlayer } from "../player.controller";
+import { ErrorType, ServiceError } from "../../util";
+
+/*eslint-disable @typescript-eslint/no-explicit-any*/
+describe("Player controller", () => {
+  let createSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    createSpy = jest.spyOn(PlayerService, "createPlayer");
+  });
+
+  afterEach(() => {
+    createSpy.mockRestore();
+  });
+
+  it("Responds with a 201 on successful player creation", async () => {
+    const req: any = {
+      query: {
+        gameCode: "6942069",
+        nickname: "DonkeyKong",
+      },
+    };
+
+    const send = jest.fn();
+
+    const status = jest.fn(() => {
+      return {
+        send,
+      };
+    });
+
+    const res: any = {
+      status,
+    };
+
+    const next: any = jest.fn();
+
+    createSpy.mockReturnValue("69eb420cg69");
+
+    await createPlayer(req, res, next);
+
+    expect(createSpy).toHaveBeenCalledTimes(1);
+
+    expect(status).toHaveBeenCalledTimes(1);
+    expect(status).toHaveBeenCalledWith(201);
+
+    expect(send).toHaveBeenCalledTimes(1);
+    expect(send).toHaveBeenCalledWith("69eb420cg69");
+  });
+  it("Responds with a 400 if the game code is invalid", async () => {
```

Comment:
Hmmm looks like Prettier hasn't make the white spacing between tests consistent 🤔 super minor but could be nice to clean that up

Reply:
Ooh shoot Prettier doing me dirty like that - I'll clean up some of them